### ISSUE TITLE
Package name is not unique

### ISSUE THREAD

**Issue by rjollos** [trac-hacks/trac-github#131]
@embray has raised the point that use of `tracext` without declaring a namespace package can lead to collisions (discussed [here](https://github.com/embray/trac-github/commit/5c15d881befc5abac2fbfb1a3279491eb5e39d80#comments)). I may have screwed up the implementation in 5c15d88 and 7275086.

> if two packages are installing the file `tracext/__init__.py` the second one to be installed will override the first one. Being that `tracext` is a namespace package it should either have no `__init__.py` at all (the new Python 3 mechanism) or if there is an `__init__.py` it should have that boilerplate for extending the package's `__path__`.

Here I've installed TracMercurial alongside trac-github:
```
$ python setup.py bdist_wheel
[...]
$ pip install dist/trac_github-2.4.dev0-py2.py3-none-any.whl
[...]
$ find pve/lib/python2.7/site-packages/tracext/ -name "*.py"
pve/lib/python2.7/site-packages/tracext/__init__.py
pve/lib/python2.7/site-packages/tracext/hg/hooks.py
pve/lib/python2.7/site-packages/tracext/hg/backend.py
pve/lib/python2.7/site-packages/tracext/hg/__init__.py
pve/lib/python2.7/site-packages/tracext/github.py
```

I don't understand this well enough, but I think we want the github code in `pve/lib/python2.7/site-packages/tracext/github`.

The changes I'll propose will yield:
```
$find pve/lib/python2.7/site-packages/tracext/ -name "*.py"
pve/lib/python2.7/site-packages/tracext/github/__init__.py
pve/lib/python2.7/site-packages/tracext/hg/hooks.py
pve/lib/python2.7/site-packages/tracext/hg/backend.py
pve/lib/python2.7/site-packages/tracext/hg/__init__.py
```

**Comment by aaugustin** [comment]
To be honest I copy-pasted from other Trac plugins and never totally figured out how the namespace package stuff works.

**Comment by rjollos** [comment]
I experimented quite a bit and would welcome better ways to accomplish these changes.

Putting the code in `tracext/github/__init__.py` isn't ideal, but I couldn't find another way that wouldn't change the Component rules. If the user activates the components from the web admin page, the following is found in trac.ini:

```
tracext.github.githubbrowser = enabled
tracext.github.githubgroupsprovider = enabled
tracext.github.githubloginmodule = enabled
tracext.github.githubmixin = enabled
tracext.github.githubpolicy = enabled
tracext.github.githubpostcommithook = enabled
```

Those rules are effective before and after the change, but if we put the classes in say `tracext/github/github.py`, different rules would be needed.

There's another problem that might require (eventually) changing the component rules. `GitHubLoginModule` shouldn't display on the Plugin admin page when `requests-oauthlib` is not installed since we specified the dependency in setup.py (5c15d88). However, it appears that `GitHubLoginModule` needs to be in it's own module for that to work correctly.

The way it's implemented now, we get a log message when the dependency isn't satisfied:
```
15:17:40 Trac[loader] DEBUG: Skipping "github.loginmodule = tracext.github:GitHubLoginModule [oauth]": DistributionNotFound: The 'requests_oauthlib>=0.5' distribution was not found and is required by the application
```

, but the Component is still displayed on the Plugin admin page:

![screen shot 2018-04-04 at 15 17 28](https://user-images.githubusercontent.com/1152089/38338456-8aa28990-381e-11e8-8200-6303bd7dac05.png)

The only way to fix that I see so far is to move `GitHubLoginModule` to it's own module, such as `tracext.github.auth`. That would require a different component rule for enabling:

```
tracext.github.auth.githubloginmodule = enabled
```

So if we take that step, we might as well split the code in `__init__.py` to several modules and just give the user guidance on upgrading for how to change the component rules. I'm going to look at whether change can be made on the Trac side before proposing those changes.

**Comment by rjollos** [comment]
Please note there is an associated pull request: #132.

**Comment by rjollos** [comment]
> There's another problem that might require (eventually) changing the component rules. `GitHubLoginModule` shouldn't display on the Plugin admin page when `requests-oauthlib` is not installed since we specified the dependency in setup.py (5c15d88). However, it appears that `GitHubLoginModule` needs to be in it's own module for that to work correctly.

I had forgotten about [#12158](https://trac.edgewall.org/ticket/12158).

**Comment by rjollos** [comment]
Changes merged in deaec86c.
